pkg/e2etest/e2e_test.go

```diff
@@ -0,0 +1,529 @@
+//go:build !race
+
+package e2etest
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"net"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/els0r/goProbe/cmd/goProbe/config"
+	"github.com/els0r/goProbe/cmd/goQuery/cmd"
+
+	// "github.com/els0r/goProbe/cmd/goQuery/commands"
+	"github.com/els0r/goProbe/pkg/capture"
+	"github.com/els0r/goProbe/pkg/capture/capturetypes"
+	"github.com/els0r/goProbe/pkg/goDB"
+	"github.com/els0r/goProbe/pkg/goDB/encoder/encoders"
+	"github.com/els0r/goProbe/pkg/goprobe/writeout"
+	"github.com/els0r/goProbe/pkg/results"
+	"github.com/els0r/goProbe/pkg/types"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/stretchr/testify/require"
+
+	slimcap "github.com/fako1024/slimcap/capture"
+	"github.com/fako1024/slimcap/capture/afpacket/afring"
+	"github.com/fako1024/slimcap/capture/pcap"
+	"github.com/fako1024/slimcap/link"
+)
+
+const testDataPath = "testdata"
+
+//go:embed testdata/*
```

Comment:
All data is stored in the repo code as compressed pcap files. `embed` automatically picks up the files and makes them available inside the test binary (which means the "standard" test files should be kept small). This means that if we find an errornuous scenario we can just drop a pcap file into the `testdata` folder, commit it and the test will automatically run it, no further action required.
In addition, the parameter `ext-pcap-data` allows to set a directory or file that contains pcap files from a local source (e.g. in order to test larger files or even huge sets of test data) manually.